### PR TITLE
feat(eslint-plugin): deprecate `no-duplicate-imports` in favour of `import/no-duplicates`

### LINKED ISSUE #4238: [no-duplicate-imports] deprecate rule in favour of `import/no-duplicates`
Separating out from https://github.com/typescript-eslint/typescript-eslint/issues/4225#issuecomment-980111513

In #2609 we added an extension rule for the core `no-duplicate-imports`. In hindsight - this wasn't a great idea because there already exists an even better rule in the ecosystem - [`import/no-duplicates`](https://github.com/import-js/eslint-plugin-import/blob/HEAD/docs/rules/no-duplicates.md).

Instead of us attempting to maintain a competing rule we should instead deprecated our version of the rule and point users at the `eslint-plugin-import` rule instead.

Similar to `camelcase` - we can maintain a markdown doc which points users to the better rule.

### PR DESCRIPTION
**Pull request by snewcomer**
## PR Checklist

-   [x] Addresses an existing open issue: fixes #4238
-   [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
-   [ ] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md) were taken

## Overview

Deprecate `no-duplicate-imports` in favor of [`import/no-duplicates](https://github.com/import-js/eslint-plugin-import/blob/HEAD/docs/rules/no-duplicates.md).

Note: checking on a few codebases that this rule does what it needs for the TS community.

### CODE REVIEW

**Comment by bradzacher** [comment]
There are two bits of config you can add to the rule metadata which will help signal the rule is deprecated. IIRC they are `deprecated` and `deprecatedBy`

File: packages/eslint-plugin/docs/rules/no-implicit-any-catch.md [hunk 1]
```diff
@@ -8,6 +8,8 @@ By default, TypeScript will type a catch clause variable as `any`, so explicitly
 
 The `noImplicitAny` flag in TypeScript does not cover this for backwards compatibility reasons, however you can use `useUnknownInCatchVariables` (part of `strict`) instead of this rule.
 
+## DEPRECATED
```
Comment (ehzhang) [inline packages/eslint-plugin/docs/rules/no-implicit-any-catch.md:11]: Hi! I noticed that this has been deprecated, but this seems like an unrelated rule to `no-duplicate-imports`

Is this intentional? If not, is there a different rule we should use?

Comment (snewcomer) [reply packages/eslint-plugin/docs/rules/no-implicit-any-catch.md:11]: https://github.com/typescript-eslint/typescript-eslint/blob/186fcf7bdee8d4cb33bc2f26b2961529f1bae493/packages/eslint-plugin/src/rules/no-implicit-any-catch.ts#L17

It was added [here](https://github.com/typescript-eslint/typescript-eslint/commit/deeb7bb9334d301c6af56aefd37d318231af11ef)

> [TS 4.4 now has a compiler flag for this](https://devblogs.microsoft.com/typescript/announcing-typescript-4-4-rc/#use-unknown-catch-variables). So it's now a rule we don't need! Because 4.4 is still in RC, we'll just deprecate it, and remove it in the next major.
